examples/automatic-labelling-from-localhost/variables.tf

```diff
@@ -28,3 +28,15 @@ variable "zone" {
   type        = string
   description = "The zone in which resources will be applied."
 }
+
+variable "network" {
+  type        = string
+  description = "The name or self_link of the network to create compute instance in. Only one of network or subnetwork should be specified."
+  default     = ""
```

Comment:
Empty defaults should ~never be used. If a default value is provided, it *must* be a usable valuable for at least some customers.

In this case, I think `default = "default"` would work.

Reply:
The problem with `default = "default”` is that the`google_compute_instance resource` expect only one of the `network`, `subnetwork` to be specified, and only `subnetwork` should be specified in case of custom network.
So, if I put` var network  { default     = "default" } and the user has custom network, the user will have to provide `subnetwork = "subSelfLink"` and it will fail due to network = `default`.  Approach with “null” defaults has been gotten from https://github.com/terraform-google-modules/terraform-google-vm/tree/master/modules/compute_instance But I found a solution, we can specify only `subnetwork` and `default = default` So, if you have a default/auto net you can go with default value, if a custom one then you have to specify subnetwork anyway.